Dev

ArtsemAverkov opened the pull request:
Test container Task

ArtsemAverkov committed:
add: Utils pacage and Gift/Order/Tags.class, rderPostgreSqlRepositoryTest and UserPostgresQLRepositoryTest fix: same test method

ArtsemAverkov committed:
add: ControllerExceptionHandlerTest and moved the methods for creating a request and response for controller, service, repository into a separate package

ArtsemAverkov committed:
add: test exceptions

ArtsemAverkov committed:
pull master

Fodense commented on src/test/java/ru/clevertec/ecl/common/mapper/serviceMapper/UserServiceImplTestMapper.java:
```diff
@@ -0,0 +1,25 @@
+    public static User builderUserDto(UserDtoRequest userDto){
+        return User.builder()
+                .id(RequestId.VALUE_1.getValue())
+                .name(userDto.getName())
```
а что ты тут проверяешь? 
не вижу ассертов

Fodense commented:
И я не нашёл тест контейнер

Fodense commented on src/test/java/ru/clevertec/ecl/common/repository/OrderPostgreSqlRepositoryTest.java:
```diff
@@ -0,0 +1,80 @@
+            .parse("postgres:14.3");
+    @Container
+    private  static final PostgreSQLContainer<?> postgresSQLContainer =
+            new PostgreSQLContainer<>(PostgresQL_IMAGE)
```
Да вижу. Окей)
Вынеси всё что связано с контейнером в отдельный класс и наследуй везде где он нужен

Fodense replied:
И вместо @Container, юзай
@BeforeAll
    static void start() {
        jdbcDatabaseContainer.start();
    }
    
    Или как-то так

Fodense replied:
Так же создай папочку integration и перемести туда все такие тесты

ArtsemAverkov committed:
changing the start method with the @Container annotation to @BeforAll,
moving the test container to the integration folder, moved everything related to the test container into a separate class and inherited it where it is needed, added Loger
